File: array.java
/****************************************************************************
 *          17th Annual UCF High School Programming Tournament              *
 *                              May 2, 2003                                 *
 *                     University of Central Florida                        *
 *                                                                          *
 *                                                                          *
 * Special thanks to our sponsors: UPE, SAIC, ACM, and HARRIS               *
 *                                                                          *
 *                                                                          *
 * Problem:         Array                                                   *
 * Problem Author:  Ben                                                     *
 * Solution Author: Mike W                                                  *
 * Data Author:     Greg                                                    *
 ***************************************************************************/
import java.io.*;
import java.util.*;

public class array
{
  /**
   *  Processes input for array.
   *  @throws IOException if problem reading input.
   */
  public static void main(String[] args) throws IOException
  {
    BufferedReader in;  // input reader
    int num;            // number in array
    StringTokenizer st; // tokenizer
        
    in = new BufferedReader(new FileReader("array.in")); // open input
        
    // data causes break;
    while(true)
    {
      // Read the size of the case.
      num = Integer.parseInt(in.readLine().trim());
            
      // terminal case.
      if(num == 0) break;
            
      // read input data
      st = new StringTokenizer(in.readLine(), " ");
            
      //Loop through all the numbers in the array
      for(int i=0;i<num;i++)
      {
        //We're looking for '1'. If found, print position and end.
        if(Integer.parseInt(st.nextToken())==1)
        {
          System.out.println("The Unity is at position " + i + ".");
          break;
        }
      }                        
      System.out.println();
    }
    in.close();
  }
}

File: array.cpp
/****************************************************************************
 *          17th Annual UCF High School Programming Tournament              *
 *                              May 2, 2003                                 *
 *                     University of Central Florida                        *
 *                                                                          *
 *                                                                          *
 * Special thanks to our sponsors: UPE, SAIC, ACM, and HARRIS               *
 *                                                                          *
 *                                                                          *
 * Problem:         Array                                                   *
 * Problem Author:  Ben                                                     *
 * Solution Author: Mike W                                                  *
 * Data Author:     Greg                                                    *
 ***************************************************************************/
#include<stdio.h>
#include<stdlib.h>

int main()
{
  FILE * in;
  int number;
  int i;
  int temp;
  char buf[30];

  in = fopen("array.in","r");

  //Process data and place a new line between each case.
  while(1)
  {
    //Scans in the number of elements in the array
    fscanf(in,"%d\n",&number);
    //if it's zero return false, because it's the ending case
    if(number==0) return 0;

    //Loop through the given array
    for(i=0;i<number;i++)
    {
      //Scan in the number
      fscanf(in,"%d",&temp);

      //If it's one print it out, but keep reading to finish line.
      if(temp==1)
      {
        printf("The Unity is at position %d.\n",i);
      }
    }

    printf("\n");
  }

  fclose(in);

  return 0;
}
